src/Data/MessagePack/Class.hs

```diff
@@ -188,6 +191,23 @@ instance MessagePack a => MessagePack [a] where
     ObjectArray xs -> mapM fromObject xs
     _              -> fail "invalid encoding for list"
 
+instance MessagePack a => MessagePack (V.Vector a) where
+  toObject = ObjectArray . map toObject . V.toList
+  fromObject = \case
+    ObjectArray o -> V.fromList <$> mapM fromObject o
```

Comment:
And here, perhaps
```haskell
fromObject o = V.fromList <$> fromObject o
-- or point-free
fromObject = fmap V.fromList . fromObject
```
The disadvantage is that the error message will be about "list" instead of "Vector", but I think that's not very important.